### ISSUE TITLE
Rename Markdown formatted hint files.

### ISSUE THREAD

**Issue by SirWumpus** [ioccc-src/winner#16]
Rename Markdown formatted hint files to `hint.md` or README.md within their respective directories so that GitHub will automatically format when viewed online.

**Comment by lcn2** [comment]
Excellent point, @SirWumpus Thank you!

We are in the process of doing a massive edit to the site.  We will do this has part of our edit.

We will keep this open as a reminder until we finish this.

**Comment by lcn2** [comment]
For historical reasons, we sometimes have both `foo.html` as well as `foo.md` (or some improper filename variant of a markdown file).  We **do not** want to keep both `HTML` and `markdown` versions of the same thing.

When it comes time for us to rebuild the web site, we are thinking of tossing or redirecting the `HTML` link into a properly named `markdown` link.

We are only talking about cases where there is BOTH, say a `foo.html` AND a `foo.html`.  For now, if there is only one of those files, we would keep it.

What do you, @SirWumpus, think about keeping only the `marldown` links in those duel file cases?

If we do remove one of the two files, should we attempt to keep a `META tag` redirect to the other file to keep the old link alive?

**Comment by SirWumpus** [comment]
Well `marldown` might make for interesting marshmallow coffee. That aside...

I'd prefer the Markdown version over the HTML, given Markdown by definition is suppose to be readable as text or formatted by a viewer, so  a/ GitHub easily renders a Markdown file, b/ in GitHub clicking on .html files shows you the raw source with no means to view it as it should be, c/ or just simply plain text when no other choice.  Personally I'd toss the .html in favour of .md /.txt versions.

BTW do not go out of the way with older winners to convert their hint / doc / txt files into Markdown.  That could take ages.  Maybe invite winners / fans to convert them (re-jigger into directories, reformat old hint files, etc.) for GitHub viewing.   Also remember some nutters like me have spoiler GitHubahuba repos online that you could "point" where its all pretty.

Anyway, while keeping the old versions makes sense for historical reasons, it might be time to update for presentation sake.  Not sure how older winners should be handled just now, but I'd say stop when you get to them and invite the fans to help.

**Comment by SirWumpus** [comment]
For older contests, it might be nice to at least rename IWON.hint to IWON.txt to make clear how it can be viewed without actually reformatting the contents.  Reformatting old winner hint files could be a rabbit hole.  Oh BTW there are a couple hint files that appear to be ROT13.  Those should be fixed IMO  since no one cares any more.

**Comment by xexyl** [comment]
> Oh BTW there are a couple hint files that appear to be ROT13. Those should be fixed IMO since no one cares any more.

This is not true though. I do care and the winners possibly do too. Also in 2020/ferguson2 there's a file that might at first glance appear to be rot13 but it's actually not. In fact there are two files - three if you count the html generated file - that might appear to be rot13 but are in enciphered with different keys to the Enigma machine itself. Those absolutely cannot change.

**Comment by xexyl** [comment]
Oh and I want to say another reason I like the rot13 in the entries that have them. Some of us (myself included) can read in blocks and it's possible (not guaranteed - it depends on how I feel) that I could scroll down and suddenly have a clearer idea what went on and if I wanted to figure it out it's suddenly that option is gone. 

So keeping rot13 is a good thing. If someone doesn't want to have their spoilers in that form that's fine but it should be left up to the author. People do care and if they didn't they wouldn't have had it in enciphered in the first place. 

I can't speak for all authors that have done this but I really enjoyed encrypting my Enigma spoilers with the Enigma machine itself. Now that's probably more fun than rot13 but still it was fun and in that case it also lets the user actually use the program. There was even a fun challenge with a delicious reward (at least if they decided to make use of the file - a recipe) in that entry for the user though not one that anyone would probably go for since the same file was also in the other winning entry that year.

Anyway that's why rot13 should not be undone - if something is enciphered there's a reason for it and spoilers is a great example.

**Comment by xexyl** [comment]
> For historical reasons, we sometimes have both `foo.html` as well as `foo.md` (or some improper filename variant of a markdown file). We **do not** want to keep both `HTML` and `markdown` versions of the same thing.
> 
> When it comes time for us to rebuild the web site, we are thinking of tossing or redirecting the `HTML` link into a properly named `markdown` link.
> 
> We are only talking about cases where there is BOTH, say a `foo.html` AND a `foo.html`. For now, if there is only one of those files, we would keep it.
> 
> What do you, @SirWumpus, think about keeping only the `marldown` links in those duel file cases?
> 
> If we do remove one of the two files, should we attempt to keep a `META tag` redirect to the other file to keep the old link alive?

I know you didn't ask me but I thought I'd bring up something that Simon brought up to me that I think is true as well and a very good point. I was going to paste it but I can't find it - not sure which email it's in. Anyway he was telling me that some of the complaints he's heard is how the entries main information is not rendered in html - that it's just text. That was in 2020 and in 2022 it's just as true that it would be easier to read the judges' comments and the author comments if it was in html.

In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.

But personally I use both - if I'm opening it in vim (which is usually how it goes) I'll look at the markdown file as well.

**Comment by lcn2** [comment]
> In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.

When we rebuild the web site, tools that can render markdown into html can be considered.  However such html file will not be part of the primary manifest, but rather built from markdown.

**Comment by xexyl** [comment]
> > In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.
> 
> When we rebuild the web site, tools that can render markdown into html can be considered. However such html file will not be part of the primary manifest, but rather built from markdown.

Personally I don't mind either way.

**Comment by lcn2** [comment]
> > > In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.
> > 
> > 
> > When we rebuild the web site, tools that can render markdown into html can be considered. However such html file will not be part of the primary manifest, but rather built from markdown.
> 
> Personally I don't mind either way.

A _repo on the side_ is being prepared whereby we will reduce information to a single file (instead of, for example, having both a markdown file and an html file).  After that in this _repo on the side_ we will generate JSON manifest files that document the primary list of files for each winning entry.

If there will be alternate formats of certain files (such as html rendered versions of markdown files), then those will alternate format files will be documented in another JSON file as a file that is "generated" (rather than a primary file).  If this is done, then we will need to write a tool to generate these alternate format files prior to uploading (via GitHub publishing) to the web site.

**Comment by xexyl** [comment]
> > > > In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.
> > > 
> > > 
> > > When we rebuild the web site, tools that can render markdown into html can be considered. However such html file will not be part of the primary manifest, but rather built from markdown.
> > 
> > 
> > Personally I don't mind either way.
> 
> A _repo on the side_ is being prepared whereby we will reduce information to a single file (instead of, for example, having both a markdown file and an html file). After that in this _repo on the side_ we will generate JSON manifest files that document the primary list of files for each winning entry.

I know you said it but literal thinker picked up on it immediately so asking. Just to be sure you mean that there won't usually be a html / markdown pair but only one file and probably html?

I guess the manifest info is not fully decided but if it's something I can help with please let me know. It can be here or in the mkiocccentry repo. Whichever works.

> 
> If there will be alternate formats of certain files (such as html rendered versions of markdown files), then those will alternate format files will be documented in another JSON file as a file that is "generated" (rather than a primary file). If this is done, then we will need to write a tool to generate these alternate format files prior to uploading (via GitHub publishing) to the web site.

Same thing goes - if there's anything I can help with please do let me know.

I hope I replied to everything for now. Going to spend a bit of time just watching a show and then go to sleep. Hope you have a great night my friend!

**Comment by lcn2** [comment]
> > > > > In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.
> > > > 
> > > > 
> > > > When we rebuild the web site, tools that can render markdown into html can be considered. However such html file will not be part of the primary manifest, but rather built from markdown.
> > > 
> > > 
> > > Personally I don't mind either way.
> > 
> > 
> > A _repo on the side_ is being prepared whereby we will reduce information to a single file (instead of, for example, having both a markdown file and an html file). After that in this _repo on the side_ we will generate JSON manifest files that document the primary list of files for each winning entry.
> 
> I know you said it but literal thinker picked up on it immediately so asking. Just to be sure you mean that there won't usually be a html / markdown pair but only one file and probably html?

Some files will be generated by a tool from primary files.  For example the web page for the winners will be generated from JSON files and a base template.

Markdown files could be used to generate HTML files from the same toolset.

> I guess the manifest info is not fully decided but if it's something I can help with please let me know. It can be here or in the mkiocccentry repo. Whichever works.

Once we finish our pass on the _repo on the side_ we will open it up to review.  However as the toolset has not been written, such generated files will be incomplete, inconsistent or missing until the toolset is working.

Help on that front, when the time comes, would be appreciated.

> > If there will be alternate formats of certain files (such as html rendered versions of markdown files), then those will alternate format files will be documented in another JSON file as a file that is "generated" (rather than a primary file). If this is done, then we will need to write a tool to generate these alternate format files prior to uploading (via GitHub publishing) to the web site.
> 
> Same thing goes - if there's anything I can help with please do let me know.
> 
> I hope I replied to everything for now. Going to spend a bit of time just watching a show and then go to sleep. Hope you have a great night my friend!

Thank you and best wishes @xexyl.

**Comment by xexyl** [comment]
> > > > > > In other words if there is a markdown file and it's feasible to generate an html file that might be a good idea.
> > > > > 
> > > > > 
> > > > > When we rebuild the web site, tools that can render markdown into html can be considered. However such html file will not be part of the primary manifest, but rather built from markdown.
> > > > 
> > > > 
> > > > Personally I don't mind either way.
> > > 
> > > 
> > > A _repo on the side_ is being prepared whereby we will reduce information to a single file (instead of, for example, having both a markdown file and an html file). After that in this _repo on the side_ we will generate JSON manifest files that document the primary list of files for each winning entry.
> > 
> > 
> > I know you said it but literal thinker picked up on it immediately so asking. Just to be sure you mean that there won't usually be a html / markdown pair but only one file and probably html?
> 
> Some files will be generated by a tool from primary files. For example the web page for the winners will be generated from JSON files and a base template.
> 
> Markdown files could be used to generate HTML files from the same toolset.
> 
> > I guess the manifest info is not fully decided but if it's something I can help with please let me know. It can be here or in the mkiocccentry repo. Whichever works.
> 
> Once we finish our pass on the _repo on the side_ we will open it up to review. However as the toolset has not been written, such generated files will be incomplete, inconsistent or missing until the toolset is working.
> 

Opening it to review sounds good to me.

> Help on that front, when the time comes, would be appreciated.
> 

Please do let me know! If I don't answer here (and you write here) please do so on the other repo perhaps in the mostly OT thread.

> > > If there will be alternate formats of certain files (such as html rendered versions of markdown files), then those will alternate format files will be documented in another JSON file as a file that is "generated" (rather than a primary file). If this is done, then we will need to write a tool to generate these alternate format files prior to uploading (via GitHub publishing) to the web site.
> > 
> > 
> > Same thing goes - if there's anything I can help with please do let me know.
> > I hope I replied to everything for now. Going to spend a bit of time just watching a show and then go to sleep. Hope you have a great night my friend!
> 
> Thank you and best wishes @xexyl.

You too my friend!

**Comment by xexyl** [comment]
> This issue is being addressed in the [temp-test-ioccc repo](https://github.com/ioccc-src/temp-test-ioccc). Once the pull request is completed against this repo, it will be implemented.

Yup .. already done in fact .. I did it in commit 851e191519953c743cdd363dc19099e89210319c on 8 February!
